components/hab/src/cli.rs

```diff
@@ -858,7 +861,8 @@ pub fn sub_sup_run() -> App<'static, 'static> {
         (@arg PEER_WATCH_FILE: --("peer-watch-file") +takes_value conflicts_with[peer]
             "Watch this file for connecting to the ring"
         )
-        (@arg RING: --ring -r +takes_value "Ring key name")
+        (@arg RING: --ring -r env(RING_ENVVAR) "Ring key name")
+        (@arg RING_KEY: --("ring-key") env(RING_KEY_ENVVAR) "Contents of the ring key")
```

Comment:
Should `RING` and `RING_KEY` be mutually exclusive?

Reply:
Yeah, I think they should. Also, I think it should be made more clear with examples what kind of values `RING` and `RING_KEY` would take on.

Reply:
The resolution logic implies that they are.

Reply:
One concern I have is that making an explicit option for this behavior encourages users to put key material directly into a command-line invocation, which may expose secrets.

(I have similar reservations our use of Builder auth token options, FWIW).

I think we could leave this fallback environment variable logic in the underlying resolution logic, without exposing it as a formal CLI option. Testing could be done using the abstractions introduced in #5840 when it merges.